test/dynamo/test_functions.py

```diff
@@ -1021,7 +1014,28 @@ def fn(x):
             x_inference = torch.randn(2, 2)
 
         cnts = torch._dynamo.testing.CompileCounter()
-        opt_fn = torch.compile(fn, backend=cnts, fullgraph=True)
+        opt_fn = torch.compile(fn, backend=cnts, fullgraph=False)
+
```

Comment:
I mentioned this offline, but there is some is_inference switching code in flash-attention: https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/layers/rotary.py#L419

Parts of this codebase are fullgraph compileable and parts aren't (I don't remember which) so we should check that we're not regressing anything here